File: sdks/java/src/main/java/dev/nucleus/NucleusClient.java
package dev.nucleus;

import dev.nucleus.admin.UsersApi;
import dev.nucleus.admin.OrgsApi;

public class NucleusClient {
    private static boolean warned = false;
    private final String secretKey;
    private final String baseUrl;

    private NucleusClient(Builder builder) {
        this.secretKey = builder.secretKey;
        this.baseUrl = builder.baseUrl;
        if (!warned) {
            String version = "0.1.0-dev.1";
            if (version.contains("dev")) {
                System.err.println("[Nucleus] WARNING: You are using a dev preview (" + version + "). Do not use in production.");
            }
            warned = true;
        }
    }

    public UsersApi users() { return new UsersApi(baseUrl, secretKey); }
    public OrgsApi organizations() { return new OrgsApi(baseUrl, secretKey); }

    public NucleusClaims verifyToken(String jwt) { return NucleusTokenVerifier.verify(jwt, baseUrl); }

    public static Builder builder() { return new Builder(); }

    public static class Builder {
        private String secretKey;
        private String baseUrl = "https://api.nucleus.dev";
        public Builder secretKey(String key) { this.secretKey = key; return this; }
        public Builder baseUrl(String url) { this.baseUrl = url; return this; }
        public NucleusClient build() { return new NucleusClient(this); }
    }
}

File: sdks/dotnet/src/Nucleus/NucleusClient.cs
using Nucleus.Admin;

namespace Nucleus;

/// <summary>
/// Main entry point for the Nucleus .NET SDK.
/// Provides token verification and admin API access.
/// </summary>
public sealed class NucleusClient : IDisposable
{
    private readonly HttpClient _httpClient;
    private readonly string _baseUrl;
    private readonly NucleusTokenVerifier _verifier;

    public NucleusClient(string secretKey, string baseUrl = "https://api.nucleus.dev")
    {
        ArgumentException.ThrowIfNullOrWhiteSpace(secretKey);

        _baseUrl = baseUrl.TrimEnd('/');
        _httpClient = new HttpClient();
        _httpClient.DefaultRequestHeaders.Authorization =
            new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", secretKey);
        _httpClient.DefaultRequestHeaders.Add("Accept", "application/json");
        _verifier = new NucleusTokenVerifier(_baseUrl);

        Users = new UsersApi(_httpClient, _baseUrl);
        Orgs = new OrgsApi(_httpClient, _baseUrl);
    }

    /// <summary>Admin Users API.</summary>
    public UsersApi Users { get; }

    /// <summary>Admin Organizations API.</summary>
    public OrgsApi Orgs { get; }

    /// <summary>
    /// Verifies a Nucleus JWT and returns the parsed claims.
    /// </summary>
    public Task<NucleusClaims> VerifyTokenAsync(string token, CancellationToken cancellationToken = default) =>
        _verifier.VerifyAsync(token, cancellationToken);

    public void Dispose()
    {
        _httpClient.Dispose();
        _verifier.Dispose();
    }
}
